app/services/rent_service.rb

```diff
@@ -0,0 +1,56 @@
+module RentService
+  def self.reservation(user, rent)
+    Rent.transaction do
+      rent.validate!
+      rent.user = user
+      rent.status = user.try(:member?) ? :confirmed : :unconfirmed
+      rent.save
+      if rent.council.present?
+        rent.overlap.each(&:overbook)
+      end
+      RentMailer.rent_email(rent).deliver_now
```

Comment:
Hmm... Look this up but I think that any exception thrown in a transaction will rollback the transaction. If this isn't what you want, you can catch the exception.